04/cesarova_sifra.py

```diff
@@ -0,0 +1,46 @@
+from string import ascii_lowercase, ascii_uppercase
+
+plaintext = input('Jaky mam zasifrovat text? ')
+ciphertext = ''
+
+while True:
+    #Jestlize odpoved neni cislo, tak se dotaz opakuje
+    key = (input('Jaky mam zadat klic? Zadej cele cislo: '))
+    if key.isnumeric():
+        key = int(key)
+        break
+
+
+for pismeno in plaintext:
+    #jestlize je pismeno v abecede
+    if pismeno in (ascii_uppercase[:26]):
+        #prevede pismeno na znak
+        znaky = (ord(pismeno)) - (ord('A'))
+        #print(znaky)
+        #prevede znak na pozadovanou sifru
+        ci = (znaky + key) % 26
+        ci = (ci + (ord('A')))
+        #print(ci)
+        ciphertext += (chr(ci))
+    elif pismeno in (ascii_lowercase[:26]):
+        #prevede pismeno na znak
+        znaky = (ord(pismeno)) - (ord('a'))
+        #print(znaky)
+        #prevede znak na pozadovanou sifru
+        ci = (znaky + key) % 26
+        ci = (ci + (ord('a')))
+        #print(ci)
+        ciphertext += (chr(ci))
```

Comment:
Algoritmus (až na ty extra závorky všude možně 😄) je správně napsaný a funguje - dobrá práce 🍀 

Ale dovolím si bonusovou poznámečku - všimni si, že všechny řádky krom dvou z každého bloku pod `if` a `elif` jsou totožé - prostě duplikované. To je silný signál pro to, že by se algoritmus dal přepsat tak, aby neobsahoval takové množství duplikovaného kódu. A to například takto:

```python
for pismeno in plaintext:
    if pismeno in ascii_uppercase[:26]:
        alphabet_start = ord('A')
    elif pismeno in ascii_lowercase[:26]:
        alphabet_start = ord('a')
    else:
        ciphertext += pismeno
        # Přeskoč zbytek těla "for"u a jdi na další písmeno.
        continue  

    # Na tomhle místě už víme, že to bylo fakt nějaké písmeno.
    letter_alphabet_position = ord(pismeno) - alphabet_start
    ci = (letter_alphabet_position + key) % 26
    ciphertext += chr(ci + alphabet_start)
```